p2p/src/net/mock/transport/channel.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use once_cell::sync::Lazy;
+use tokio::sync::mpsc;
+
+use crate::net::mock::{types::Message, SocketService, TransportService};
+
+static NETWORK_HANDLE: Lazy<mpsc::UnboundedSender<SocketCommand>> = Lazy::new(|| {
+    let (tx, mut rx) = mpsc::unbounded_channel();
+    tokio::spawn(async move {
+        loop {
```

Comment:
I know this is a draft but this can probably but moved into a separate function at some point

Reply:
Maybe even an actual `MockNetwork` object